src/tests/sqlsmith/src/sql_gen/query.rs

```diff
@@ -102,33 +105,132 @@ impl<'a, R: Rng> SqlGenerator<'a, R> {
             select_list,
             from,
             selection,
-            // TODO
-            group_by: None,
+            group_by,
             // TODO
             having: None,
             // TODO
             window_list: None,
         }
     }
 
-    fn gen_select_list(&mut self) -> Vec<SelectTarget> {
-        let select_num = self.rng.gen_range(1..=5);
-        let mut targets = Vec::with_capacity(select_num);
-        for _ in 0..select_num {
-            let target = match self.rng.gen_range(0..=9) {
-                0..=9 => {
-                    let ty = self.gen_data_type();
-                    let expr = self.gen_expr(&ty);
-                    SelectTarget::AliasedExpr {
-                        expr: Box::new(expr),
-                        // TODO
-                        alias: None,
+    fn gen_group_by(&mut self) -> Option<GroupBy> {
+        let mode = self.rng.gen_range(0..=5);
+        let group_cap = self.rng.gen_range(1..=5);
+        let mut groupby_items = Vec::with_capacity(group_cap);
+
+        for _ in 0..group_cap {
+            let ty = self.gen_data_type();
+            let expr = self.gen_expr(&ty);
+            let groupby_item = match expr {
+                Expr::Literal {
+                    lit: Literal::UInt64(_),
+                    ..
+                } => Expr::Cast {
+                    span: None,
+                    expr: Box::new(expr),
+                    target_type: TypeName::UInt64,
+                    pg_style: false,
+                },
+                Expr::Literal {
+                    lit:
+                        Literal::Decimal256 {
+                            scale, precision, ..
+                        },
+                    ..
+                } => {
+                    if scale == 0 {
+                        Expr::Cast {
+                            span: None,
+                            expr: Box::new(expr),
+                            target_type: TypeName::Decimal { precision, scale },
+                            pg_style: false,
+                        }
+                    } else {
+                        expr
                     }
                 }
-                // TODO
-                _ => unreachable!(),
+                _ => expr,
             };
-            targets.push(target)
+            groupby_items.push(groupby_item);
+        }
+
+        match mode {
```

Comment:
The probability of generating a group by is somewhat 80%, which I think is a bit too high, we only need a 20% or 30% probability of generating a group by, that's enough.